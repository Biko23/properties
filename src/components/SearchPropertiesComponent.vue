<template>
<div>
    <v-container id="container" fluid>
        <v-row id="property-header">
            <div style="flex:1;">
                <h3>Properties</h3>
                <small style="font-weight: bold;">{{allSearchedResults.length}} results</small>
            </div>
            <div style="flex:1;">
                <h3 style="color: #3b6ef3;">BUY PROPERTY HERE</h3>
            </div>
        </v-row>
        <v-row id="main-property">
            <v-col cols="12" xl="2" lg="3" md="4" sm="6" xs="12" v-for="(currentProperty, index) in allSearchedResults" :key="index">
                <property-card
                    :location="currentProperty.name"
                    :date="formatDate(currentProperty.when_created)"
                    :cost="currentProperty.actual_value"
                    :postedBy="currentProperty.created_by"
                    :src="'http://localhost:8002/' + currentProperty.snapshot"
                    :to="`/view/${currentProperty.property_id}?location=${currentProperty.name}`"
                />
            </v-col>
        </v-row>
    </v-container><br />
</div>
</template>

<script>
import PropertyCard from '@/components/PropertyCard'
import dateFormat from 'dateformat'
import { mapActions, mapGetters } from 'vuex';
export default {
    name: "SearchPropertiesComponent",
    components: {
        PropertyCard
    },
    props: ["keyword"],
    methods: {
        ...mapActions(["fetchPropertiesBySearchKeyword"]),
        formatDate(dateToFormat){
            let currentDate = new Date();
            let result;
            const formattedDate = dateFormat(currentDate, "isoDateTime");
                console.log("formatedDate", formattedDate, "dateToFormat", dateToFormat);
            if(dateToFormat == formattedDate || dateToFormat == (formattedDate - 1)){
                result = dateFormat(dateToFormat, "DDDD");
            } else {
                result = dateFormat(dateToFormat, "dddd, mmmm dS, yyyy");
            }
            return result;
        },
        logKeyword(){
            console.log(this.keyword);
        }
    },
    computed: {
        ...mapGetters(["allSearchedResults"]),
    },
    created(){
        this.fetchPropertiesBySearchKeyword();
    }
};
</script>

<style scoped>
#property-header {
    margin: 0 10px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
#main-property {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}
</style>
