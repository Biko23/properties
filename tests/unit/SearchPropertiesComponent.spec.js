import { createLocalVue, mount } from '@vue/test-utils'
import SearchPropertiesComponent from '@/components/SearchPropertiesComponent.vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueRouter from 'vue-router'


describe('SearchPropertiesComponent.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)
    let actions
    let getters
    let store
    let vuetify
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
        actions = {
            fetchPropertiesBySearchKeyword: jest.fn(),
            fetchAutoCompleteWords: jest.fn(),
            fetchFavoritePropertiesForComparision: jest.fn(),
            removePropertyFromFavorites: jest.fn(),
            addPropertyToFavorites: jest.fn(),
            postAUserLog: jest.fn()
        }
        getters = {           
            allSearchedResults: () => ({}),
            allAutocompletedList: () => ({}),
            currentLoggedinUser: () => ({}),
            allCurrentUserFavoriteProperties: () => ({}),
            loginState: () => ({})
        }
        store = new Vuex.Store({
            actions,
            getters
        })
    })

    const mountFactory = options => {
        return mount(SearchPropertiesComponent, {
            localVue,
            vuetify,
            store,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const commaFormatted = jest.fn()
        const wrapper = mountFactory({
            methods: { commaFormatted }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })
})

