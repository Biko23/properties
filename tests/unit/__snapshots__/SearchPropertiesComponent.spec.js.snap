// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPropertiesComponent.vue should have a match snapshot 1`] = `
<div>
  <div class="container container--fluid" id="container">
    <div role="dialog" class="v-dialog__container">
      <!---->
    </div>
    <div id="property-header" class="row">
      <div id="result-total">
        <h3>Properties for Sale</h3> <small style="font-weight: bold;"> results</small>
      </div>
      <div id="search-field" style="position: relative;">
        <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--filled v-text-field--enclosed v-text-field--outlined v-text-field--placeholder v-select v-autocomplete">
          <div class="v-input__control">
            <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-5" class="v-input__slot white">
              <fieldset aria-hidden="true">
                <legend><span>â€‹</span></legend>
              </fieldset>
              <div class="v-select__slot"><input autofocus="autofocus" id="input-5" placeholder="Search Properties" type="text" autocomplete="off"><input type="hidden"></div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div> <button type="button" class="v-icon notranslate mr-2 v-icon--link mdi mdi-magnify theme--light" style="font-size: 40px; color: blue; position: absolute; right: 0px; top: 10%; z-index: 100;"></button>
      </div>
    </div>
    <div id="main-property" class="row"></div>
  </div><br>
</div>
`;
